src/mail-app/mail/model/MailModel.ts

```diff
@@ -114,29 +114,49 @@ export class MailModel {
 		const mailboxDetails = await this.mailboxModel.getMailboxDetails()
 
 		const tempFolders = new Map<Id, MailboxSets>()
+		let lastCaughtRestError: TutanotaError | null = null
 
 		for (let detail of mailboxDetails) {
-			if (detail.mailbox.folders) {
-				const mailSets = await this.loadMailSetsForListId(neverNull(detail.mailbox.folders).folders)
+			const foldersRef = detail.mailbox.folders
+			if (foldersRef != null) {
+				let mailSets: MailFolder[]
+				try {
+					mailSets = await this.loadMailSetsForListId(foldersRef.folders)
+				} catch (e) {
+					if (e instanceof NotAuthorizedError || e instanceof NotFoundError) {
+						console.warn("Got", e.name, "when trying to load a mailbox", detail.mailbox._id, "(cached user likely outdated)")
+					} else {
+						throw e
+					}
+
+					lastCaughtRestError = e
```

Comment:
can we move these two lines into the first branch of the condition? i got mighty confused and was thinking that `lastCaughtRestError` declaration is incorrect. If we have just two branches and nothing after it's easier to follow